src/dmd/root/filename.d

```diff
@@ -894,4 +895,82 @@ version(Windows)
 
         return F(ret);
     }
+
+}
+
+version(Posix)
+{
+
+    /**************************
+    Takes a callable F and applies it to the result of converting
+    `fileName` to an absolute file path (char*)
+    Params:
+        fileName = The file name to be converted to an absolute path
+    Returns: Whatever `F` returns.
+    */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        import core.sys.posix.stdlib: realpath;
+        char[1024] realPathBuffer;
+        auto absPath = realpath(fileName, &realPathBuffer[0]);
+        return F(absPath);
+    }
+}
+else
+{
+    /**************************
+    Takes a callable F and applies it to the result of converting
+    `fileName` to an absolute file path (char*)
+    Params:
+        fileName = The file name to be converted to an absolute path
+    Returns: Whatever `F` returns.
+     */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        import core.sys.windows.winnls: WideCharToMultiByte;
+        import core.stdc.stdlib: malloc, free;
+        import std.stdio: writeln;
+        import std.conv: text;
+
+        return fileName.extendedPathThen!((wpath) {
+                // first find out how long the buffer must be to store the result
+                const length = WideCharToMultiByte(0,    // code page
+                                                   0,    // flags
+                                                   wpath,
+                                                   -1,   // wpath len, -1 is null terminated
+                                                   null, // multibyte output ptr
+                                                   0,    // multibyte output length
+                                                   null, // default char
+                                                   null, // if used default char
+                );
+
+                if(!length) return F((char*).init);
+
+                char[1024] buf;
+
+                scope multibyteBuf = length > buf.length
+                    ? (cast(char*)malloc(length * char.sizeof))
+                    : &buf[0];
```

Comment:
@ other reviews: I have seen this stack buffer / malloc buffer pattern now three times this week. Maybe it's time to abstract this?

With std.experimental.allocator it would be simply:

```d
StackFront!(1024, Mallocator.instance);
```

See also: https://dlang.org/phobos/std_experimental_allocator_showcase.html

Reply:
Makes sense, but this is a bug fix targeting stable.

Reply:
@wilzbach : Maybe it's just a good pattern ? :)
There have been talk of making `scope foo = new char[](runtime_length);` work, which is exactly what we need.

Reply:
Hehe my point was just that this "good pattern" could be reduced from ~10 lines to a single line with sth. like Phobos's `StackFront`

> There have been talk of making scope foo = new char[](runtime_length); work,

Yep, that would be great too (: